res/admin-roles.py

```
@@ -0,0 +1,417 @@
+    params = {"pageSize": page_size}
+    if name is not None:
+        params["name"] = name
+    if role_type is not None:
+        params["type"] = ROLE_TYPES[role_type]
+
+    roles = []
+    current = 0
+    while True:
+        current += 1
+        params["current"] = current
+        response = requests.get(
```

Ruff (0.16.3)

[error] Probable use of `requests` call without timeout

(S113)